### PR TITLE
Added Introduction playgrounds examples

### PR DESCRIPTION
**Pull request by paweldudek**

### CODE REVIEW

File: Introduction.playground/Pages/2. Transforming Sequences.xcplaygroundpage/Contents.swift [hunk 1]
```diff
@@ -0,0 +1,33 @@
+//: [Previous](@previous)
+
+import Foundation
+import RxSwift
+
+//: # Transforming sequences
+//:
+//: In this part we'll learn how to transform sequences. The most basic transform operator that's defined in RxSwift is `map`, which you probably already know from `Swfit.Sequence`.
```
Comment (siejkowski) [inline Introduction.playground/Pages/2. Transforming Sequences.xcplaygroundpage/Contents.swift]: Swfit -> Swift

File: Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift [hunk 2]
```diff
@@ -0,0 +1,53 @@
+//: [Previous](@previous)
+
+import Foundation
+import RxSwift
+
+//: # Memory Management - Part Two
+//:
+//: In previous example we learned that certain sequence observers are automatically disposed when a sequence completes.
```
Comment (siejkowski) [inline Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift]: It's just a case of wording, but I think it's important to put emphasis on the fact that *every* subscription *always* disposes when *any* observable emits onCompleted on onError. It's not just sequence observers, it's one of the invariants of the Rx framework

File: Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+//:
+//: You probably wonder - what if I have a sequence that never completes? Or a sequence that won't complete before my object should be deallocated?
+//: In this case we need to manually dispose of the sequence observer. There are two ways of doing this.
```
Comment (siejkowski) [inline Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift]: *manually* suggests that the `dispose` is called by ourselves. Which is true, but only in the first case. Maybe `In this case the subscription disposal must be taken care of by ourselves`?

File: Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+// Another way of disposing, one that is much more elegant, is using a `DisposeBag`.
+var exampleDisposeBag = DisposeBag()
+
+// You can add a disposable to dispose bag by calling
+disposable.disposed(by: exampleDisposeBag)
+
+// In order to dispose of all registered sequence observers within a dispose bag you just need to deallocate dispose bag. You can do so by, for instance, replacing it with a new value:
+exampleDisposeBag = DisposeBag()
+
```
Comment (siejkowski) [inline Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift:34]: It's an awesome explanation, I'd just add that `DisposeBag` is mostly used to bind the lifecycle of subscription to lifecycle of the object within which the subscription takes place

File: Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+//:
+//: ## Assignment
+//: 
+//: Your task is to dispose of a registered observer using a dispose bag. In this scenario we'll use a sequence that does not complete immedtiately - a `Variable`. `Variable` stores a value within and sends an event each time you update that value by calling `variable.value = x`.
```
Comment (siejkowski) [inline Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift]: immedtiately -> immediately

File: Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift (hunk 4 above)
Comment (literator) [reply Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift:34]: Agree. That's quite important in real cases.

**Review by literator — APPROVED** [review]

File: Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift (hunk 3 above)
Comment (paweldudek) [reply Introduction.playground/Pages/6. Memory Management - Part Two.xcplaygroundpage/Contents.swift]: I think that 

> In this case we need to take care how the sequence observer is disposed of. There are two ways of doing this.

fits this the best.

**Comment by paweldudek** [comment]
@siejkowski @literator I've updated this PR accordingly. Please review changes.

Thanks!

**Comment by siejkowski** [comment]
It's ok for me, ready to merge 👍

**Comment by literator** [comment]
👍

--
Maciej Oczko

**Pull request merged**